### PR TITLE
chore(network-chaos): Adding docker runtime in litmus lib

### PR DESCRIPTION
**Pull request by ispeakc0de**
Signed-off-by: shubhamchaudhary <shubham.chaudhary@mayadata.io>

### CODE REVIEW

File: experiments/generic/pod-network-loss/experiment/pod-network-loss.go [hunk 1]
```diff
@@ -101,7 +101,7 @@ func PodNetworkLoss(clients clients.ClientSets) {
 	}
 
 	// Including the pumba lib for pod-network-loss
-	if experimentsDetails.ChaosLib == "litmus" && experimentsDetails.ContainerRuntime == "docker" {
+	if experimentsDetails.ChaosLib == "pumba" && experimentsDetails.ContainerRuntime == "docker" {
```
Comment (uditgaurav) [inline experiments/generic/pod-network-loss/experiment/pod-network-loss.go:104]: Do you want to make this change for others as well? That is for corruption, duplication, latency, and loss?

File: experiments/generic/pod-network-loss/experiment/pod-network-loss.go [hunk 2]
```diff
@@ -114,7 +114,7 @@ func PodNetworkLoss(clients clients.ClientSets) {
 		}
 		log.Info("[Confirmation]: The pod network loss chaos has been applied")
 		resultDetails.Verdict = "Pass"
-	} else if experimentsDetails.ChaosLib == "litmus" && (experimentsDetails.ContainerRuntime == "containerd" || experimentsDetails.ContainerRuntime == "crio") {
+	} else if experimentsDetails.ChaosLib == "litmus" {
```
Comment (uditgaurav) [inline experiments/generic/pod-network-loss/experiment/pod-network-loss.go:117]: same as above for others as well?

File: build/litmus-go/Dockerfile [hunk 3]
```diff
@@ -23,6 +23,12 @@ RUN curl -L https://github.com/alexei-led/pumba/releases/download/${PUMBA_VERSIO
 #Installing promql cli binaries
 RUN curl -L https://github.com/litmuschaos/test-tools/raw/master/custom/promql-cli/promql-linux-${TARGETARCH} --output /usr/local/bin/promql && chmod +x /usr/local/bin/promql
 
+RUN if [[${TARGETARCH} == "arm64"]]; then \
+    curl -L https://get.docker.com/builds/Linux/aarch64/docker-1.13.1.tgz --output docker-cli && tar -zxvf docker-cli docker/docker && mv docker/docker /usr/local/bin/docker && chmod +x /usr/local/bin/docker; \
+  else \
+    curl -L https://get.docker.com/builds/Linux/x86_64/docker-1.13.1.tgz --output docker-cli && tar -zxvf docker-cli docker/docker && mv docker/docker /usr/local/bin/docker && chmod +x /usr/local/bin/docker; \
+  fi
```
Comment (uditgaurav) [inline build/litmus-go/Dockerfile:30]: ref: https://github.com/litmuschaos/litmus-go/pull/224/files#r539612380

**Review by uditgaurav — APPROVED** [review]

File: chaoslib/litmus/network-chaos/helper/netem.go [hunk 4]
```diff
@@ -132,68 +136,118 @@ loop:
...
-		if container.Name == experimentDetails.TargetContainer {
-			containerID = strings.Split(container.ContainerID, "//")[1]
-			break
+	switch experimentDetails.ContainerRuntime {
+	case "docker":
+		// deriving the container id of the pause container
+		cmd := "docker ps | grep k8s_POD_" + experimentDetails.TargetPods + "_" + experimentDetails.AppNS + " | awk '{print $1}'"
+		out, err := exec.Command("/bin/sh", "-c", cmd).CombinedOutput()
+		if err != nil {
+			log.Error(fmt.Sprintf("[cri]: Failed to run docker ps command: %s", string(out)))
```
Comment (ksatchit) [inline chaoslib/litmus/network-chaos/helper/netem.go:149]:
```suggestion
			log.Error(fmt.Sprintf("[docker]: Failed to run docker ps command: %s", string(out)))
```

File: chaoslib/litmus/network-chaos/helper/netem.go [hunk 5]
```diff
@@ -132,68 +136,118 @@ loop:
...
+//GetPID extract out the PID of the target container
+func GetPID(experimentDetails *experimentTypes.ExperimentDetails, containerID string) (int, error) {
+	var PID int
+
+	switch experimentDetails.ContainerRuntime {
+	case "docker":
+		// deriving pid from the inspect out of target container
+		out, err := exec.Command("docker", "inspect", containerID).CombinedOutput()
+		if err != nil {
+			log.Error(fmt.Sprintf("[cri]: Failed to run docker inspect: %s", string(out)))
```
Comment (ksatchit) [inline chaoslib/litmus/network-chaos/helper/netem.go:183]:
```suggestion
			log.Error(fmt.Sprintf("[docker]: Failed to run docker inspect: %s", string(out)))
```

File: chaoslib/litmus/network-chaos/helper/netem.go [hunk 6]
```diff
@@ -132,68 +136,118 @@ loop:
...
+			return 0, err
+		}
+		// parsing data from the json output of inspect command
+		PID, err = parsePIDFromJSON(out, experimentDetails.ContainerRuntime)
+		if err != nil {
+			log.Error(fmt.Sprintf("[cri]: Failed to parse json from docker inspect output: %s", string(out)))
```
Comment (ksatchit) [inline chaoslib/litmus/network-chaos/helper/netem.go:189]:
```suggestion
			log.Error(fmt.Sprintf("[docker]: Failed to parse json from docker inspect output: %s", string(out)))
```

**Review by ksatchit — APPROVED** [review]

**Review by ksatchit — APPROVED** [review]

**Pull request merged**